### PR TITLE
Add item to agenda regarding module ordering

### PR DESCRIPTION
**Pull request by bmeck**

### CODE REVIEW

**Comment by devsnek** [comment]
FWIW my thoughts on this are below:

```js
// a.js
import './b';
import './c';

// b.js
import 'nonexistant';

// c.js
import { mount } from 'fuse';

// assume this is a sync method
mount('/program');
```

if `c.js` evaluates before `b.js`, whatever device this runs on will get completely wrecked by fuse.

- `b.js` could also contain something like `throw new Error('XYZ');`, but it doesn't matter, in either case there is no way to handle that in `c.js`
- ordered evaluation doesn't fix this, it gives a false sense of security, exposing mounting to `a.js` would be the most responsible thing to do.

My next thought came to my mind is speed, mostly in terms of load time.

Consider the following example:

```js
// app has a really big dep graph
import app from './app';
// loading has a really small dep graph
import { stop } from './loading';

app.on('ready', stop);
```

Yes this can be reordered if there is evaluation order but then you've got devs thinking about the optimal order of their imports, and probably after a few months some libraries will pop up for finding the best order to put your imports in so that your app loads fast.

**Having evaluation order either does not affect this example or hinders it.**

There are two reasons to enforce evaluation order that I can think of:

1. Predictability

"I know that `b.js` will evaluate before `c.js`" is a nice thing to say, and I don't think there is any direct argument against predictability. However I don't think that means it automatically wins. Being able to choose when modules evaluates is a powerful feature, but that plays more into point two.

2. Discouraging use of old module systems with custom loaders (I think the v8 devs touched on this point in a google doc somewhere)

I think this is a double edged sword. I speak from a node.js perspective here. As a user I want to see libraries move to ESM. They could be written to work in node and browser at the same time with the same code without any webpacking or magic, and that's really really cool. However, I don't think pushing users into a new language feature just because someone thinks it's better (and it might be better) is the right thing to do. It also gives up a lot of control that someone with a custom loader would have. I can't imagine everything that someone might want to do with a loader, but I don't want to shut down the possibility because I don't have a use case.

TL;DR Imposed loading order from the spec is a bad idea and should be left up to the loaders.
